test/TextToBinary.SubgroupDispatch.cpp

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2016 Google
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and/or associated documentation files (the
+// "Materials"), to deal in the Materials without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Materials, and to
+// permit persons to whom the Materials are furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be included
+// in all copies or substantial portions of the Materials.
+//
+// MODIFICATIONS TO THIS FILE MAY MEAN IT NO LONGER ACCURATELY REFLECTS
+// KHRONOS STANDARDS. THE UNMODIFIED, NORMATIVE VERSIONS OF KHRONOS
+// SPECIFICATIONS AND HEADER INFORMATION ARE LOCATED AT
+//    https://www.khronos.org/registry/
+//
+// THE MATERIALS ARE PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+// CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
+
+// Assembler tests for instructions in the "Barrier Instructions" section
+// of the SPIR-V spec.
+
+#include "UnitSPIRV.h"
+
+#include "TestFixture.h"
+#include "gmock/gmock.h"
+
+namespace {
+
+using ::testing::AllOf;
+using ::testing::Eq;
+using ::testing::Matcher;
+using ::testing::Property;
+using ::testing::SizeIs;
+using std::vector;
+
+// Creates a matcher of SPIR-V word vectors that matches a single instruction
+// with the given opcode and length. TODO(dekimir): move this into TestFixture
+// and DRY other tests that perform this match in longhand.
+Matcher<vector<uint32_t>> IsOpcode(SpvOp opcode, uint16_t length) {
```

Comment:
The number of layers is actually quite low, and the benefit substantial: this lets us check a `CompiledInstructions` result (with meaningful failure messages) in the shortest, most readable, least error-prone fashion.  Other tests either have multiple lines to assert the vector length and the first element (assuming they even test both), or they rely on `ElementsAre(spvOpcodeMake(...), _, _, _, _)`, which gets cumbersome with 7-argument instructions like the ones tested here.

I'll change the name to something like `IsSingleInstruction`.